File: test/llvm_swift_test_code/taint_analysis/taint_exception_03.swift
@_silgen_name("source")
func source() -> Int

@_silgen_name("sink")
func sink(_ p: Int)

struct S {
    var data: Int
    init(_ data: Int) {
        self.data = data
    }
}
@main
struct MyMain {
    static func main() {
        var a = source()
        sink(a)
        var s = S(0)  // original C++ code was: S *s = new S(0);
        var b = a
        sink(b)
    }
}

File: test/llvm_test_code/taint_analysis/dummy_source_sink/taint_exception_03.cpp
extern int source();
extern void sink(int data);

struct S {
  int data;
  S(int data) : data(data) {}
};

int main() {
  int a = source();
  sink(a);
  S *s = new S(0);
  int b = a;
  sink(b);
  return 0;
}
